User/Library/UserMisc.c

```diff
@@ -672,3 +674,23 @@ DevicePathFromHandle (
   return NULL;
 }
 
+VOID
+EFIAPI
+Sha512TransformAvx (
+  IN OUT UINT64      *State,
+  IN     CONST UINT8 *Data,
+  IN     UINTN       BlockNb
+  )
+{
+  ASSERT (FALSE);
+}
+
+BOOLEAN
+EFIAPI
+TryEnableAvx (
+  VOID
+  )
+{
+  mIsAvxEnabled = FALSE;
+  return FALSE;
+}
```

Comment:
Is the asm code actually plumbed in, in this version? I would have expected to see a real and a dummy implementation of this method in the ASM files, and no implementation here, and similar for variable `mIsAvxEnabled`.
(cc @mhaeuser )

Reply:
@MikeBeaton Vice-versa, I think we want this C code in the lib over the dummy ASM: https://github.com/acidanthera/OpenCorePkg/pull/252#discussion_r614914319

Reply:
I saw the comment, hence the cc to yourself - was not expecting that link would work with > 1 implementation - possibly I'm just not aware of how this is getting resolved.

Trying to compile and run this, have spotted that multiple utilities are also using Sha2.c (hence currently linking with the .c implementation, so definitely not using this acceleration). I guess as part of final commit, we'd want to make the tools link to this new implementation too?

Reply:
Apologies, I see now that the versions here _are_ the versions designed to link into the utilities.

However, would still want to suggest (@vit9696 @mhaeuser ?) that it might be ideal to link the accelerated version into the utilities, too.

Reply:
@MikeBeaton cli will throw an exception, imo not worth the effort

Reply:
Okay, will take @mhaeuser's steer that using this outside OC itself is more trouble than it's worth, in which case nothing else to add on this one.